crates/noirc_evaluator/src/ssa/conditional.rs

```diff
@@ -0,0 +1,485 @@
+            ctx.remove_block(i);
+        }
+        Ok(())
+    }
+
+    pub fn conditionalize_block(&self, ctx: &mut SsaContext, block: BlockId) {
+        let assumption_id = ctx[block].assumption;
+        let assumption = &self[assumption_id];
+        let ass_value = assumption.value.unwrap();
+
+        if ass_value == ctx.zero() {
+            todo!();
+        }
+        let block_kind = ctx[block].kind.clone();
+        let instructions = ctx[block].instructions.clone();
+        for i in &instructions {
+            let ins1 = ctx.get_instruction(*i);
+            let ins = ins1.clone();
+            match &ins.operation {
+                Operation::Phi { block_args, .. } => {
+                    if block_kind == BlockType::IfJoin {
+                        assert_eq!(block_args.len(), 2);
+                        let ins2 = ctx.get_mut_instruction(*i);
+                        ins2.operation = Operation::Cond {
+                            condition: assumption.condition,
+                            val_true: block_args[0].0,
+                            val_false: block_args[1].0,
+                        }
+                    }
+                }
+
+                Operation::Store { array_id, index, value } => {
+                    if !ins.operation.is_dummy_store() && ass_value != ctx.one() {
+                        let load = Operation::Load { array_id: *array_id, index: *index };
+                        let e_type = ctx.mem[*array_id].element_type;
+                        let dummy =
+                            ctx.add_instruction(Instruction::new(load, e_type, Some(block)));
+                        let pos = ctx[block]
+                            .instructions
+                            .iter()
+                            .position(|value| *value == ins.id)
+                            .unwrap();
+                        ctx[block].instructions.insert(pos - 1, dummy);
+                        let operation = Operation::Cond {
+                            condition: ass_value,
+                            val_true: *value,
+                            val_false: dummy,
+                        };
+                        let cond =
+                            ctx.add_instruction(Instruction::new(operation, e_type, Some(block)));
+                        ctx[block].instructions.insert(pos, cond);
+                        //store the conditional value
+                        let ins2 = ctx.get_mut_instruction(*i);
+                        ins2.operation =
+                            Operation::Store { array_id: *array_id, index: *index, value: cond };
+                    }
+                }
+                Operation::Intrinsic(_, _) => {
+                    if ass_value != ctx.one() {
+                        if let ObjectType::Pointer(a) = ins.res_type {
+                            let array = &ctx.mem[a].clone();
+                            let name = array.name.to_string() + "_dup";
```

Comment:
Can you make "_dup" a constant string, with an explanation as to why its needed.

Eg: ` const DUPLICATED : &str = "_dup"`